src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java

```diff
@@ -1781,16 +1805,18 @@ protected static void waitBatcherCompleted(Batcher<RevisionInternal> b) {
         }
     }
 
-    protected static void waitPendingFuturesCompleted(BlockingQueue<Future> futures) {
+    protected void waitPendingFuturesCompleted() {
         try {
-            while (!futures.isEmpty()) {
-                Future future = futures.take();
+            while (!pendingFutures.isEmpty()) {
+                Future future = pendingFutures.take();
                 try {
                     future.get();
                 } catch (InterruptedException e) {
                     Log.e(Log.TAG_SYNC, "InterruptedException in Future.get()", e);
                 } catch (ExecutionException e) {
                     Log.e(Log.TAG_SYNC, "ExecutionException in Future.get()", e);
+                } finally {
+                    runnables.remove(future);
```

Comment:
`Runnable` should be removed when task completed.  It should not be removed before `waitPendingFuturesCompleted()`.

Unprocessed `Runnable`s will be canceled and removed from runnables in `stop()` or `close()` methods